mobile-sdk/src/main/java/com/github/itwin/mobilesdk/ITMApplication.kt

```diff
@@ -592,38 +594,56 @@ open class ITMApplication(
         webView?.let { webView ->
             (webView.parent as? ViewGroup)?.removeView(webView)
             container.addView(webView)
-            webView.setOnApplyWindowInsetsListener { v, insets ->
-                updateSafeAreas(v)
-                v.onApplyWindowInsets(insets)
+            webView.setOnApplyWindowInsetsListener { view, insets ->
+                updateSafeAreas(view, insets)
+                view.onApplyWindowInsets(insets)
             }
         }
     }
 
-    private fun updateSafeAreas(view: View) {
-        val activity = ((view.parent as? ViewGroup)?.context as? Activity) ?: return
-        val window = activity.window ?: return
-        val message = mutableMapOf(
-            "left" to 0.0f,
-            "right" to 0.0f,
-            "top" to 0.0f,
-            "bottom" to 0.0f
-        )
-        window.decorView.rootWindowInsets?.displayCutout?.let { displayCutoutInsets ->
+    private data class Quint<out A, out B, out C, out D, out E>(
+        val first: A,
+        val second: B,
+        val third: C,
+        val fourth: D,
+        val fifth: E
+    ) {
+        /**
+         * Returns string representation of the [Quint] including its [first], [second], [third], [fourth], and [fifth] values.
+         */
+        override fun toString(): String = "($first, $second, $third, $fourth, $fifth)"
+    }
+
+    private fun getSafeAreas(activity: Activity, view: View, insets: WindowInsets): Quint<Float, Float, Float, Float, Float> {
+        fun toDp(value: Int): Float {
             val density = activity.resources.displayMetrics.density
-            val top = displayCutoutInsets.safeInsetTop / density
-            val bottom = displayCutoutInsets.safeInsetBottom / density
-            val left = displayCutoutInsets.safeInsetLeft / density
-            val right = displayCutoutInsets.safeInsetRight / density
-            // Make both sides have the same safe area.
-            val sides = max(left, right)
-            message["left"] = sides
-            message["right"] = sides
-            // Include the actual left/right insets so developers can use them if desired.
-            message["minLeft"] = left
-            message["minRight"] = right
-            message["top"] = top
-            message["bottom"] = bottom
+            return value / density
         }
+        var insetsType = WindowInsetsCompat.Type.displayCutout()
+        // Ideally we'd always get system bars insets but before API 30 getInsets() returns bars insets even if they are overlaid on top.
+        if (activity.isInMultiWindowMode)
+            insetsType = insetsType or WindowInsetsCompat.Type.systemBars()
+
+        val safeAreaInsets = WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType)
+        return safeAreaInsets.let {
+            Quint(toDp(max(it.left, it.right)), toDp(it.left), toDp(it.right), toDp(it.top), toDp(it.bottom))
+        }
```

Comment:
I know the code came from Field but the use of `let` here seems odd and unnecessary to me. It seems like they used `let` just to make the parameters to the `Quint` constructor shorter. The same could be accomplished by renaming `safeAreaInsets` to something shorter:

```suggestion
        val in = WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType)
        return Quint(toDp(max(in.left, in.right)), toDp(in.left), toDp(in.right), toDp(in.top), toDp(in.bottom))
```

Reply:
Another possible option:

```suggestion
        return WindowInsetsCompat.toWindowInsetsCompat(insets, view).getInsets(insetsType).let {
            Quint(toDp(max(it.left, it.right)), toDp(it.left), toDp(it.right), toDp(it.top), toDp(it.bottom))
        }
```

Reply:
I'm going to instead get rid of the variable entirely and use `with()`.

Reply:
Done: as an aside, the Quint was all me. The SYNCHRO code only used sides, top, and bottom, so they used a (built-in) Triple. I created the Quint class to add left and right to the return value to support the minLeft and minRight fields that they didn't include.